eng/publish-coverlet-result-files.yml

```diff
@@ -0,0 +1,44 @@
+steps:
+- task: Powershell@2
+  displayName: Prepare coverage files to Upload
+  inputs:
+    targetType: inline
+    pwsh: true
+    script: |
+      New-Item -ItemType Directory "$(Build.SourcesDirectory)/artifacts/TestLogs/"
+      function Copy-FileKeepPath {
+      param (
+          $filter,$FileToCopy,$des,$startIndex
+          )
+      Get-ChildItem -Path $FileToCopy -Filter $filter -Recurse -File | ForEach-Object {
+          $fileName = $_.FullName
+          #Remove the first part to ignore from the path.
+          $newdes=Join-Path -Path $des -ChildPath $fileName.Substring($startIndex)
+          $folder=Split-Path -Path $newdes -Parent
+          $err=0
+
+          #check if folder exists"
+          $void=Get-Item $folder -ErrorVariable err  -ErrorAction SilentlyContinue
```

Comment:
I'm not that familiar with azure devops pipelines but what is this `$void` variable for? In general this seems a bit complicated to me. What about
```powershell
      if (!(Test-Path -Path $folder)) {New-Item $folder -Type Directory -Force -Verbose}
      Copy-Item -Path $fileName -Destination $newdes -Recurse -Force -Verbose
```

Reply:
$void is used to suppress some PSScriptAnalyzer warnings. I like your 2 lines but there is a small deatsil. The folder is not created if it already exists. I will check this with a separte commit.